purchase_discount/models/stock_move.py

```diff
@@ -7,6 +7,11 @@
 class StockMove(models.Model):
     _inherit = "stock.move"
 
+    def write(self, values):
+        if self.env.context.get("skip_update_price_unit"):
```

Comment:
Do not you check that price_unit come in values?

Reply:
Also poping price_unit if it is the only value modified or do nothing if not